lib/ansible/modules/network/nxos/nxos_gir.py

```diff
@@ -224,8 +222,14 @@ def get_commands(module, state, mode):
             commands.append('no system mode maintenance timeout {0}'.format(
                             module.params['system_mode_maintenance_timeout']))
 
-    elif module.params['system_mode_maintenance_shutdown'] is True:
+    elif (module.params[
+        'system_mode_maintenance_shutdown'] is True and
+            mode == 'normal'):
```

Comment:
how about `elif (module.params['system_mode_maintenance_shutdown'] and mode == 'normal'):`?